R/diffdf.R

```diff
@@ -194,12 +188,42 @@ diffdf <- function(
         msg = "COMP is missing variables specified in KEYS"
     )
 
-    assertthat::assert_that(
-        all(!(KEYS %in% exclude_cols)),
-        msg = "KEYS are either an invalid or contain different modes between BASE and COMP"
+
+    assert_valid_keys <- function(COMPARE, KEYS, component, msg) {
```

Comment:
I feel like its bad practice to embed a function within a function.

Also we could then test this function individually